dev/UndockedRegFreeWinRT/UndockedRegFreeWinRT-AutoInitializer.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Windows.Foundation.UndockedRegFreeWinRTCS
+{
+    internal static class NativeMethods
+    {
+        [DllImport("Microsoft.WindowsAppRuntime.dll", CharSet = CharSet.Unicode, ExactSpelling = true)]
+        internal static extern int UndockedRegFreeWinRT_EnsureIsLoaded();
+    }
+
+    class AutoInitialize
+    {
+        [global::System.Runtime.CompilerServices.ModuleInitializer]
+        internal static void AccessWindowsAppSDK()
+        {
+            int hr = NativeMethods.UndockedRegFreeWinRT_EnsureIsLoaded();
+            if (hr < 0)
+            {
+                global::System.Environment.Exit(hr);
```

Comment:
I believe in failure cases, this may result in an app appearing to _do nothing_ and confuse users. It may also cause a (temporary) startup loop for packaged apps. (I haven't verified that, but I recall failures during packaged app startup result in retry logic.)

Reply:
Perhaps this should throw so WER gets a whack at it? And you get event logging, telemetry, etc.?

Reply:
This is only used when an app has been built using WinAppSDK/SelfContained. The only thing this really does is `LoadLibrary(Microsoft.WindowsAppRuntime.dll)`. The only way that could fail is if the application's missing that file, or something it imports (FrameworkUDK.dll - also SelfContained and placed with the app - and some OS DLLs (kernel32.dll etc)). So while it's somewhat a moral equivalent to MddBootstrapAutoInitializer.cs it's a ton simpler.

I considered baking out a rich palette of options like `MddBootstrapInitialize2(...options)` but that seemed way overkill. There's no lookup, no registered packages, and no real logic (`UndockedRegFreeWinRT_EnsureIsLoaded()` is just `{return S_OK}`). Given this is only in play when an app's using WinAppSDK/SelfContained failures here seem exceedingly rare.

I could change this to just call the function, ignore the return code and use the throwing form of P/Invoke. If the DLL can't load it would throw a .NET exception about a missing native DLL? Been a while since I saw that. Would that give a clearer symptom of the problem?

Reply:
If the DllImport's target DLL isn't found or otherwise won't load .NET throws a `DllNotFoundException`. This is similar to the C++ experience where an imported DLL can't be resolved when called loads.

`UndockedRegFreeWinRT_EnsureIsLoaded()` does nothing (`{return S_OK}`) so there's no need for any error check. The error check is if .NET can resolve the DLL, which the DllImport handles. Streamlined the code to match.